packages/contracts-bedrock/scripts/DeployOPChain.s.sol

```diff
@@ -282,41 +285,125 @@ contract DeployOPChainOutput {
     // -------- Assertions on chain architecture --------
 
     function assertValidDeploy(DeployOPChainInput _doi) internal view {
-        // TODO Add other assertions from ChainAssertions.sol here
         assertValidSystemConfig(_doi);
+        assertValidL1CrossDomainMessenger(_doi);
+        assertValidL1StandardBridge(_doi);
+        assertValidOptimismMintableERC20Factory(_doi);
+        assertValidOptimismPortal(_doi);
+        assertValidDisputeGameFactory(_doi);
+        assertValidDelayedWETHs(_doi);
+        // TODO Other FP assertions like the dispute games, anchor state registry, etc.
+        // TODO add initialization assertions
     }
 
     function assertValidSystemConfig(DeployOPChainInput _doi) internal view {
-        // forgefmt: disable-start
-        require(systemConfigProxy().owner() == _doi.systemConfigOwner(), "SC10");
-        require(systemConfigProxy().basefeeScalar() == _doi.basefeeScalar(), "SC20");
-        require(systemConfigProxy().blobbasefeeScalar() == _doi.blobBaseFeeScalar(), "SC30");
-        require(systemConfigProxy().batcherHash() == bytes32(uint256(uint160(_doi.batcher()))), "SC40");
-        require(systemConfigProxy().gasLimit() == uint64(30000000), "SC50");// TODO allow other gas limits?
-        require(systemConfigProxy().unsafeBlockSigner() == _doi.unsafeBlockSigner(), "SC60");
-        require(systemConfigProxy().scalar() >> 248 == 1, "SC70");
+        SystemConfig systemConfig = systemConfigProxy();
+
+        DeployUtils.assertInitialized({ _contractAddress: address(systemConfig), _slot: 0, _offset: 0 });
+
+        require(systemConfig.owner() == _doi.systemConfigOwner(), "SC-10");
+        require(systemConfig.basefeeScalar() == _doi.basefeeScalar(), "SC-20");
+        require(systemConfig.blobbasefeeScalar() == _doi.blobBaseFeeScalar(), "SC-30");
+        require(systemConfig.batcherHash() == bytes32(uint256(uint160(_doi.batcher()))), "SC-40");
+        require(systemConfig.gasLimit() == uint64(30000000), "SC-50"); // TODO allow other gas limits?
+        require(systemConfig.unsafeBlockSigner() == _doi.unsafeBlockSigner(), "SC-60");
+        require(systemConfig.scalar() >> 248 == 1, "SC-70");
 
         IResourceMetering.ResourceConfig memory rconfig = Constants.DEFAULT_RESOURCE_CONFIG();
```

Comment:
nit, prefer camelcasing with `rConfig` or `rsrcConfig`

Reply:
For reference this was the name used in ChainAssertions, but I agree with you, so renamed in 86d43ea80